### ISSUE TITLE
Document reproducible builds

### ISSUE THREAD

**Issue by htgoebel** [pyinstaller/pyinstaller#1714]
Please remember documenting reproducible builds. see  #1668. Thanks.

**Comment by tallforasmurf** [comment]
python -c 'import random as R ; R.seed() ; print(R.randint(0,2))'

? Is it important to set PYTHONHASHSEED to 0, 1 or 2? why not `R.randint(0,4294967295)` (as long as it is recorded for re-use)?

**Comment by tallforasmurf** [comment]
Hmmm. What am I missing here? OS X command line.

    echo $PYTHONHASHSEED
    1
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    4866
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    278
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    5334

**Comment by codewarrior0** [comment]
The random hash seed is not the random seed.

The random hash seed only comes into play with the `hash()` function (and consequently, sets and dicts which rely on the hash function). Try running `python -c 'print(hash("Hello World"))'` a few times with PYTHONHASHSEED set and unset, and with python 2.7 and 3.3 to see what the difference is.

**Comment by codewarrior0** [comment]
Actually, I can just show you:
```
$ echo 'print(hash("Hello World"))' > phs.py
$ /c/python27/python phs.py
-973091937

$ /c/python27/python phs.py
-973091937

$ PYTHONHASHSEED=1 /c/python27/python phs.py
-133323119

$ PYTHONHASHSEED=1 /c/python27/python phs.py
-133323119

$ /c/python34/python phs.py
-7283063376155732847

$ /c/python34/python phs.py
-3788090655346638451

$ PYTHONHASHSEED=1 /c/python34/python phs.py
3328880125071188731

$ PYTHONHASHSEED=1 /c/python34/python phs.py
3328880125071188731
```

With python 3.4, the hash function returns something different each time the program is run. This is a security feature to prevent attackers from being able to predict the behavior of things like dicts and sets. PYTHONHASHSEED disables this feature and makes the hash function have the same results each run.

**Comment by tallforasmurf** [comment]
Hmmm. What am I missing here? OS X command line.

    echo $PYTHONHASHSEED
    1
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    4866
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    278
    python -c 'import random; random.seed(); print(random.randint(1,10000))'
    5334

**Comment by tallforasmurf** [comment]
Sorry, I misunderstood the example code by @htgoebel in the original issue comments.
I believe that 66ee281 covers the issue.

**Comment by htgoebel** [comment]
@tallforasmurf  66ee281 is fine. (I added some suggestions, though.) Thanks!

Sorry for not reacting on your discussion earlier, I've been bussy.

**Comment by htgoebel** [comment]
Closed by 66ee281.

**Comment by tallforasmurf** [comment]
You wanted to say something about being sure you saved the hash seed value. I thought about adding something about that, but the 66ee281 text already says "known integer value", which I think tells that story to any programmer. Also, I think probably almost anybody who does this will just set it to 1. There is no benefit to using a different hash seed for different apps or different versions. I think everybody will just use 1, which is easy and also unforgettable.
